### PR TITLE
OTA improvements - DFU manager, OTARequestorDriverImpl, user response

### PR DESCRIPTION
**Pull request by ATmobica**
#### Problem
Some OTA improvements:
1. Add common DFU manager - DFU API which can be used in each example application, Now the MCUboot and CHIP OTA are supported inside.
2. Implement a custom OTA requestor driver - it allows pass callback function for the update available and apply. 
3. Add confirm and reject user response to buttons service in OTA requestor app - user can confirm or abort OTA steps.

#### Change overview
Add common DFU manager and add to OTA requestor app
Add custom OTA requestor driver implementation
Add software version string to update description - OTA requestor class
Add user response to OTA requestor app - confirm or reject update download/apply
Change log type in OTAImageProcessorImpl

#### Testing
Buil and run Mbed OTA requestor app

### CODE REVIEW

File: src/include/platform/OTARequestorDriver.h [hunk 1]
```diff
@@ -35,6 +35,7 @@ struct UpdateDescription
 {
     CharSpan imageURI;
     uint32_t softwareVersion;
+    CharSpan softwareVersionStr;
```
Comment (pan-) [inline src/include/platform/OTARequestorDriver.h:38]: This is a non owning view, is there a guaranty that it won't outlive the response ?

**Review by pan- — COMMENTED** [review]
Thanks Artur, the change looks good. I left few comments.

File: src/platform/GenericOTARequestorDriver.h [hunk 2]
```diff
@@ -53,7 +53,7 @@ class GenericOTARequestorDriver : public OTARequestorDriver
     void UpdateSuspended(System::Clock::Seconds32 delay) override;
     void UpdateDiscontinued() override;
 
-private:
+protected:
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply src/platform/GenericOTARequestorDriver.h:56]: OTARequestrorDriverImpl class inherit from this class and we need access to these class members and methods from it. I changed it to protected and set as friend class for OTARequestrorDriverImpl

File: src/platform/mbed/OTARequestorDriverImpl.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+{
+    VerifyOrDie(mRequestor != nullptr);
+    if (onOtaUpdateAvailableCallback)
+    {
+        if (!onOtaUpdateAvailableCallback->mCall(onOtaUpdateAvailableCallback->mContext, update))
+        {
+            return;
+        }
+    }
+    ScheduleDelayedAction(UpdateFailureState::kDownloading, delay,
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply src/platform/mbed/OTARequestorDriverImpl.cpp:44]: I preferred to use the CHIP solution in this case. To be compatible with OTA implementation at all.

File: src/include/platform/OTARequestorDriver.h (hunk 1 above)
Comment (ATmobica) [reply src/include/platform/OTARequestorDriver.h:38]: No, the description update object is created locally in the OTA query image response callback and uses pointers to response elements, so everything is go away with the response object. But... UpdateAvailable() callback is called inside the OTA query image response callback, so there is a guarantee in this case.

**Review by pan- — APPROVED** [review]

**Pull request merged**